### ISSUE TITLE
Dangerous builds with MPI and nlist.tree() in `next`

### ISSUE THREAD

**Issue by jglaser** [glotzerlab/hoomd-blue#563]
## Description

I get dangerous neighbor list builds when I change `nlist.cell()` to `nlist.tree()` in the lj liquid benchmark and run with two or more MPI ranks.

## Script

Check out the `next` branch of `hoomd-benchmarks`. https://github.com/glotzerlab/hoomd-benchmarks/tree/next

Change `nlist.cell()` to `nlist.tree()` in `lj_liquid/operations.py`.

Execute with below command.

## Output

```
$ mpirun -np 2 python project.py run -o lj_liquid-benchmark-gpu_np1 -f n 100
HOOMD-blue v2.8.1-509-g70a8091d3 CUDA (10.2) DOUBLE HPMC_MIXED MPI SSE SSE2 
Compiled: 12/14/2019
Copyright (c) 2009-2019 The Regents of the University of Michigan.
HOOMD-blue is running on the following GPU(s):
Rank 0:  [0]          Quadro GV100  80 SM_7.0 @ 1.63 GHz, 32508 MiB DRAM, DIS, MNG
Rank 1:  [1]          Quadro GV100  80 SM_7.0 @ 1.63 GHz, 32508 MiB DRAM, MNG
-----
You are using HOOMD-blue. Please cite the following:
* J A Anderson, C D Lorenz, and A Travesset. "General purpose molecular dynamics
  simulations fully implemented on graphics processing units", Journal of
  Computational Physics 227 (2008) 5342--5359
* J Glaser, T D Nguyen, J A Anderson, P Lui, F Spiga, J A Millan, D C Morse, and
  S C Glotzer. "Strong scaling of general-purpose molecular dynamics simulations
  on GPUs", Computer Physics Communications 192 (2015) 97--107
-----
HOOMD-blue is using domain decomposition: n_x = 1 n_y = 1 n_z = 2.
1 x 1 x 2 local grid on 1 nodes
notice(2): Group "all" created containing 1000000 particles
-----
You are using tree neighbor lists. Please cite the following:
* M P Howard, J A Anderson, A Nikoubashman, S C Glotzer, and A Z
  Panagiotopoulos. "Efficient neighbor list calculation for molecular simulation
  of colloidal systems using graphics processing units", Computer Physics
  Communications 203 (2016) 45--52
* M P Howard, A Statt, F Madutsa, T M Truskett, and A Z Panagiotopoulos.
  "Quantized bounding volume hierarchies for neighbor search in molecular
  simulations on graphics processing units", Computational Materials Science 164
  (2019) 139--146
-----
notice(2): -- Neighborlist exclusion statistics -- :
notice(2): Particles with 0 exclusions             : 1000000
notice(2): Neighbors included by diameter          : no
notice(2): Neighbors excluded when in the same body: no
** starting run **
notice(2): nlist: Dangerous neighborlist build occurred. Continuing this simulation may produce incorrect results and/or program crashes. Decrease the neighborlist check_period and rerun.
notice(2): nlist: Dangerous neighborlist build occurred. Continuing this simulation may produce incorrect results and/or program crashes. Decrease the neighborlist check_period and rerun.
```

## Expected output

No warnings. I don't get warnings with `nlist.cell()`.

## Configuration

Platform:
- Linux
- GPU (2x V100, but also observed on 2x AMD MI50 with `hip` branch)

Installation method:
- Compiled from source

Versions:
- Python version: 3.7
- **HOOMD-blue** version: `next`, 70a8091d3344a7ff3562481e27a7d338af5537b5
(**-D ENABLE_CUDA=ON -D ENABLE_MPI=ON**)
- OpenMPI 4.0.2

## Developer

possibly

**Comment by jglaser** [comment]
Any ideas @mphoward ?

**Comment by joaander** [comment]
Fixed in #564
